src/client/Client.cc

```diff
@@ -12631,13 +12631,33 @@ int Client::_rename(Inode *fromdir, const char *fromname, Inode *todir, const ch
     else
       return -EROFS;
   }
+
   if (fromdir != todir) {
     Inode *fromdir_root =
       fromdir->quota.is_enable() ? fromdir : get_quota_root(fromdir, perm);
     Inode *todir_root =
       todir->quota.is_enable() ? todir : get_quota_root(todir, perm);
-    if (fromdir_root != todir_root) {
-      return -EXDEV;
+
+    if (todir_root->quota.is_enable() && fromdir_root != todir_root) {
+      if (todir_root->quota.max_bytes &&
+         (fromdir->rstat.rbytes + todir_root->rstat.rbytes)
```

Comment:
why checking rstat of fromdir instead the inode that is being renamed.

Reply:
`fromdir` is the source inode being renamed. `fromdir_root` is the quota root of `fromdir`.

I think this checking should be right?

Reply:
file 'fromname' in fromdir is the inode being renamed

Reply:
Sorry, Yan. I totally get confused. Could you pls explain this a little more? Which inode I should use to check rstat? fromname is just string?

Reply:
Oh, I see you point